scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,113 @@ def visit_functiondef(self, node):
     # this line makes it work for async functions
     visit_asyncfunctiondef = visit_functiondef
 
+class CheckEnum(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-enum"
+    priority = -1
+    msgs = {
+        "C4746": (
+            "The enum must use uppercase naming. "
+            "https://azure.github.io/azure-sdk/python_design.html#enumerations",
+            "enum-must-be-uppercase",
+            "Capitalize enum name.",
+        ),
+        "C4747":(
+            "The enum must inherit from CaseInsensitiveEnumMeta. "
+            "https://azure.github.io/azure-sdk/python_implementation.html#extensible-enumerations",
+            "enum-must-inherit-case-insensitive-enum-meta",
+            "Inherit CaseInsensitiveEnumMeta.",
+        ),
+    }
+    options = (
+        (
+            "ignore-enum-must-be-uppercase",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not be capitalized.",
+            },
+        ),
+        (
+            "ignore-enum-must-inherit-case-insensitive-enum-meta",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not inherit CaseInsensitiveEnumMeta.",
+            },
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckEnum, self).__init__(linter)
+
+    def visit_classdef(self, node):
+        """Visits every enum class.
+
+        :param node: ast.ClassDef
+        :return: None
+        """
+        try:
+
+            inherits_metaclass, enum_class = self._check_is_enum_class(node)
+            
+            # If it has a metaclass, and is an enum class, check the capitalization
+            if inherits_metaclass and enum_class:
+                self._enum_uppercase(node)   
+            # Else if it does not have a metaclass, but it is an enum class
+            # Check both capitalization and throw pylint error for metaclass
+            elif enum_class:
+                self.add_message(
+                    "enum-must-inherit-case-insensitive-enum-meta", node=node, confidence=None
+                )
+                self._enum_uppercase(node)  
+
+        except Exception:
+            logger.debug("Pylint custom checker failed to check enum.")
+            pass
+
+    def _check_is_enum_class(self,node):
+        """Checks if a class is an enum class.
+        Returns booleans for the two enum guidelines.
+
+        :param node: ast.ClassDef
+        :return: A tuple of booleans. The first boolean will be True if the class inherits CaseInsensitiveEnumMeta.
+        The second boolean will be True if the node is an enum class.
+        :rtype: tuple[bool,bool]
+            
+        """
+        if node._metaclass:
```

Comment:
small thing, but I recall there being a `metaclass()` function on the class node? It's a little risky accessing a `_` prefixed variable since, for python, this is intended to be private. It would be safer to access the public API here if possible.